kibana-reports/server/routes/utils/reportHelper.ts

```diff
@@ -95,58 +102,104 @@ export const createVisualReport = async (
     });
   }
 
-  const timeCreated = new Date().toISOString();
-  const fileName = getFileName(name, timeCreated) + '.' + reportFormat;
-
-  //TODO: Add header and footer, phase 2
-
+  const curTime = new Date();
+  const timeCreated = curTime.valueOf();
+  const fileName = getFileName(reportName, curTime) + '.' + reportFormat;
   await browser.close();
+
   return { timeCreated, dataUrl: buffer.toString('base64'), fileName };
 };
 
 export const createReport = async (
-  report: any,
-  client: IClusterClient | IScopedClusterClient
-): Promise<{ timeCreated: string; dataUrl: string; fileName: string }> => {
+  report: ReportSchemaType,
+  client: IClusterClient | IScopedClusterClient,
+  savedReportId?: string
+): Promise<{ timeCreated: number; dataUrl: string; fileName: string }> => {
   let createReportResult: {
-    timeCreated: string;
+    timeCreated: number;
     dataUrl: string;
     fileName: string;
   };
 
-  //TODO: create new report instance with pending status
-
-  const reportSource = report.report_source;
+  // create new report instance or update saved report instance with "pending" state
+  const timePending = Date.now();
+  const saveParams: RequestParams.Index = {
+    index: CONFIG_INDEX_NAME.report,
+    id: savedReportId,
+    body: {
+      ...report,
+      state: REPORT_STATE.pending,
+      time_created: timePending,
+    },
+  };
 
-  if (reportSource === REPORT_TYPE.savedSearch) {
-    // TODO: Add createDataReport(report)
-    console.log('placeholder for createDataReport');
-  } else if (
-    reportSource === REPORT_TYPE.dashboard ||
-    reportSource === REPORT_TYPE.visualization
-  ) {
-    createReportResult = await createVisualReport(report);
+  const esResp = await callCluster(client, 'index', saveParams);
+  const reportId = esResp._id;
+
+  const reportDefinition = report.report_definition;
+  const reportParams = reportDefinition.report_params;
+  const reportSource = reportParams.report_source;
+  try {
+    if (reportSource === REPORT_TYPE.savedSearch) {
+      // TODO: Add createDataReport(report)
+      console.log('placeholder for createDataReport');
```

Comment:
I think having the comment is enough, no need for a `console.log` statement here

Reply:
Chen's PR will remove this.